File: Basics of Computing and Programming/callcoster.py
def main():
    h = {"Mon": 1, "Tue": 2, "Wed": 3, "Thr": 4, "Fri": 5, "Sat": 6, "Sun": 7}
    day = h[input("Enter the day the call started at: ")]
    time = input("Enter the time the call started at (hhmm): ")
    length = int(input("Enter the duration of the call (in minutes): "))
    if day == 6 or day == 7:
        print(f"This call will cost ${0.15 * length:.2f}")
        return
    if time >= "0800" and time <= "1800":
        print(f"This call will cost ${0.4 * length:.2f}")
    else:
        print(f"This call will cost ${0.25 * length:.2f}")
    
if __name__ == "__main__":
    main()

File: Advanced Programming in C++/callcoster.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
    map <string, int> h;
    h["Mon"] = 1;
    h["Tue"] = 2;
    h["Wed"] = 3;
    h["Thr"] = 4;
    h["Fri"] = 5;
    h["Sat"] = 6;
    h["Sun"] = 7;
    int time;
    int len;
    int day;
    double cost;
    // define and intialize paraments

    string tmp;
    cout << "Enter the day the call started at: ";
    cin >> tmp;
    day = h[tmp];
    cout << "Enter the time the call started at (hhmm): ";
    cin >> time;
    cout << "Enter the duration of the call (in minutes): ";
    cin >> len;
    // input the paraments

    if (day >= 1 && day <= 5) {
        cost = (time >= 800 && time <= 1800 ? 0.4 : 0.25) * len;
    }
    else {
        cost = len * 0.15;
    }
    // calculate the cost

    printf("This call will cost $%.2f\n", cost);
    // output the cost
    return 0;
}